client/lib/plugins/utils.js

```diff
@@ -460,3 +460,24 @@ export const getPluginPurchased = ( plugin, purchases, isMarketplaceProduct ) =>
 		)
 	);
 };
+
+/**
+ * Gets the SaaS redirect URL of a plugin if it exits and is valid
+ *
+ * @param {plugin} plugin The plugin object  to read the SaaS redirect url from
+ * @param {number} userId The user id
+ * @param {number} siteId The site id
+ * @returns The URL of the SaaS redirect page or null if it doesn't exist or is an invalid URL
+ */
+export function getSaasRedirectUrl( plugin, userId, siteId ) {
+	if ( ! plugin.saas_landing_page ) {
+		return null;
+	}
```

Comment:
Nitpicking: Add `?` would avoid it to break if no plugin is passed.

```suggestion
	if ( ! plugin?.saas_landing_page ) {
		return null;
	}
```

Reply:
I agree as this is a utility function and it is good to cope with edge cases. Committed